project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module

```diff
@@ -685,3 +691,75 @@ function c4m_og_get_user_group_nodes_by_bundle(
   }
   return $groups;
 }
+
+/**
+ * Implements hook_node_update().
+ */
+function c4m_og_node_update($node) {
+  if ($node->type != 'group') {
+    return;
+  }
+
+  c4m_og_open_group_access_activate_pending_members($node);
+}
+
+/**
+ * Activate pending members when access level of a group is changed.
+ *
+ * When changing the access level from 'Moderated' to 'Open', all pending
+ * members should become active group members.
+ *
+ * @param object $node
+ *   The node object from the "hook_node_update".
+ */
+function c4m_og_open_group_access_activate_pending_members($node) {
+  require_once DRUPAL_ROOT . '/sites/all/modules/contrib/og/includes/actions/set_state.action.inc';
+
+  $wrapper = entity_metadata_wrapper('node', $node);
+  $original_wrapper = entity_metadata_wrapper('node', $node->original);
+
+  // Group access level has been changed from 'Moderated' to 'Open'.
+  if ($wrapper->field_membership_open_request->value() == C4M_OG_OPEN_GROUP && $original_wrapper->field_membership_open_request->value() == C4M_OG_MODERATED_GROUP) {
+    $context['state'] = OG_STATE_ACTIVE;
+
+    // Activate all pending members of this group.
+    foreach (c4m_og_get_all_group_members_by_state($wrapper->getIdentifier(), OG_STATE_PENDING) as $membership_uid) {
+      $membership = og_membership_load($membership_uid);
+      og_set_state_action($membership, $context);
+    }
+  }
+}
+
+/**
+ * Get the group membership entity by user, group and state.
+ *
+ * @param int $gid
+ *   The group ID.
+ * @param int $state
+ *   The user current state.
+ *
+ * @return array
+ *   Array of members UIDs, or an empty array.
+ */
+function c4m_og_get_all_group_members_by_state($gid, $state) {
```

Comment:
always pass the $group_type as-well -- `unction c4m_og_get_all_group_members_by_state($group_type, $group_id, $state) {`